### DISCUSSION TITLE
Multi-database tenancy

### DISCUSSION THREAD

**Discussion by icaliman** [filamentphp/filament#7444] [Help]
Hi everyone!

I have a question regarding tenancy in v3.

Is it possible to have multi-database tenancy like in [stancl/tenancy](https://tenancyforlaravel.com/docs/v3/multi-database-tenancy)?

If it's possible, it would be nice to have documentation how to migrate a project from stancl/tenancy.

Regards

**Comment by pboivin** [comment]
I've never really tried that myself, but I know a few others are currently exploring it. There are a few questions on Discord already. Maybe someone will write a guide eventually. Maybe it could be you :)

**Comment by mark-nicepants** [comment]
I've got this working on an existing installation (in the process of replacing nova).

I added this middleware:

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Stancl\Tenancy\Middleware\InitializeTenancyBySubdomain;

class InitializeTenantByDomain extends InitializeTenancyBySubdomain
{
    public function handle($request, Closure $next)
    {
        $host = $request->getHost();

        // If the host is a central domain, we don't need to do anything.
        if (collect(config('tenancy.central_domains'))->contains($host)) {
            return $next($request);
        }

        return parent::handle($request, $next);
    }
}
```

To the Filament AdminPanelProvider (isPersisitent must be enabled for livewire to also be tenant aware)

```php
...
->middleware([InitializeTenantByDomain::class], isPersistent: true)
...
```


In tenancy.php config file I disabled the `asset_helper_tenancy`. This was rewriting the filament js and css asset paths.

Hope this helps someone in the future!
